src/log.c

```diff
@@ -100,16 +98,41 @@ static int fi_read_value(const char *env_name, const char * const names[])
 void fi_log_init(void)
 {
 	struct fi_filter subsys_filter;
-	int level, i;
+	int level, i, ret;
+	char *levelstr = NULL, *provstr = NULL, *subsysstr = NULL;
+
+	ret = fi_var_register(&core_prov, "log_level",
+		"Specify logging level: warn, trace, info, debug (default: warn)");
+
+	ret += fi_var_get_str(&core_prov, "log_level", &levelstr);
+
+	if (ret == -FI_ENOENT)
+		fprintf(stderr, "failed to initialize logging level from FI_LOG_LEVEL\n");
 
-	level = fi_read_value("FI_LOG_LEVEL", log_levels);
+	level = fi_read_value(levelstr);
 	if (level >= 0)
 		log_mask = ((1 << (level + 1)) - 1);
 
-	fi_create_filter(&prov_log_filter, "FI_LOG_PROV");
+	ret = fi_var_register(&core_prov, "log_prov",
+		"Specify specific provider to log (default: all)");
+
+	ret += fi_var_get_str(&core_prov, "log_prov", &provstr);
```

Comment:
same here - don't add return values.
and kill the extra blank line between assigning the value and checking with the if statement